neurots/extract_input/from_neurom.py

```diff
@@ -14,92 +14,184 @@
+        pop (neurom.core.population.Population): The given population.
 
     Returns:
-        dict: A dictionary whose structure depends on the type of distribution:
-
-        * if `type == "norm"`:
+        dict: A dictionary with the following structure:
 
         .. code-block:: bash
 
             {
-                "norm": {
-                    "mean": <mean value>,
-                    "std": <std value>
-                }
+                "size": <the soma size>
             }
+    """
+    # Extract soma size as a normal distribution
+    # Returns a dictionary with the soma information
+    soma_size = nm.get("soma_radius", pop)
+    params = stats.fit(soma_size, distribution="norm").params
+    return {"size": {"norm": {"mean": params[0], "std": params[1]}}}
 
-        * if `type == "uniform"`:
 
-        .. code-block:: bash
+def _step_fit_prob_function(angle, scale, rate):
+    """Probablity function to use for fitting angle distributions."""
 
-            {
-                "uniform": {
-                    "min": <min value>,
-                    "max": <max value>
-                }
-            }
+    def _prob(_angle):
+        return prob_function(_angle, [scale, rate], form="step") * np.sin(_angle)
 
-        * if `type == "expon"`:
+    return _prob(angle) / quad(_prob, 0, np.pi)[0]
 
-        .. code-block:: bash
 
-            {
-                "expon": {
-                    "loc": <loc value>,
-                    "lambda": <lambda value>
-                }
-            }
+def _double_step_fit_prob_function(angle, scale_low, rate_low, scale_high, rate_high):
+    """Probablity function to use for fitting angle distributions."""
+
+    def _prob(_angle):
+        return prob_function(
+            _angle, [scale_low, rate_low, scale_high, rate_high], form="double_step"
+        ) * np.sin(_angle)
+
+    return _prob(angle) / quad(_prob, 0, np.pi)[0]
+
+
+def get_fit_prob_function(morph_class="PC", neurite_type=nm.BASAL_DENDRITE, params=None):
+    """Get probability function for trunk angles.
+
+    Args:
+        morph_class (str): morphological class (PC or IN)
+        neurite_type (nm.NeuriteType): type of neurite to consider
+        params (dict): if not None, can be used to overwrite the fit functions/bounds
     """
-    if opt_distr.type == "norm":
-        return {"norm": {"mean": opt_distr.params[0], "std": opt_distr.params[1]}}
-    elif opt_distr.type == "uniform":
-        return {
-            "uniform": {
-                "min": opt_distr.params[0],
-                "max": opt_distr.params[1] + opt_distr.params[0],
-            }
-        }
-    elif opt_distr.type == "expon":
-        return {"expon": {"loc": opt_distr.params[0], "lambda": 1.0 / opt_distr.params[1]}}
-    return None
+    # this is the form of the expected parameter dict.
+    # The bounds correspond to the function's parameter min/max bounds for the fit
+    default_params = {
+        "PC": {
```

Comment:
Why do we need specific sets of parameters in the "distribution" part? Also where do we get these from if the default is not used. maybe this can be simplified?